internal/controller/git/git.go

```diff
@@ -95,6 +95,19 @@ type repo struct {
 	currentBranch string
 }
 
+// CloneOptions represents options for cloning a git repository.
+type CloneOptions struct {
+	// Branch is the name of the branch to clone. If not specified, the default
+	// branch will be cloned.
+	Branch string
+	// SingleBranch indicates whether the clone should be a single-branch clone.
+	SingleBranch bool
+	// Shallow indicates whether the clone should be with a depth of 1. This is
+	// useful for speeding up the cloning process when all we care about is the
+	// latest commit from a single branch.
+	Shallow bool
```

Comment:
Since I only see CloneOptions used one place, is it true that this PR makes it such that all clones performed by kargo are shallow?